torch/_meta_registrations.py

```diff
@@ -307,10 +307,17 @@ def calc_conv_nd_return_shape(
                 )
         return ret_shape
 
-    def pick_memory_format():
-        if input_tensor.is_contiguous(memory_format=torch.channels_last):
-            return torch.channels_last
-        elif input_tensor.is_contiguous(memory_format=torch.contiguous_format):
+    def is_channels_last(ten):
+        return ten.is_contiguous(memory_format=torch.channels_last)
+
+    def pick_memory_format(device_hint):
+        if device_hint == "cuda":
+            if is_channels_last(input_tensor) and is_channels_last(weight):
```

Comment:
I see cuda producing channels-last if only weight is channels last
```
In [8]: m=m.cuda().to(memory_format=torch.channels_last)

In [9]: inp=inp.cuda()

In [10]: m(inp).stride()
Out[10]: (968, 1, 88, 8)

In [11]: inp.stride()
Out[11]: (144, 36, 6, 1)
```

Reply:
Yea, it's either, misread the `||` as an `&&`:

https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/ConvUtils.h#L316

Reply:
Which means that all 1x1 convs will happily return channels-last format :-(

Reply:
Which makes this PR correct at least :-)